server/src/middleware.ts

```diff
@@ -39,3 +40,23 @@ export function versionedMiddleware(
   req.versioned = new VersionedMethods(req);
   next();
 }
+
+/**
+ * If the app has a configured HOST_URL, redirect any request at another host
+ * to the configured host. For example, if `HOST_URL == "https://getmyvax.org"`,
+ * this will redirect "www.getmyvax.org/x/y/z" to "getmyvax.org/x/y/z".
+ */
+export function redirectToPrimaryHost(
+  request: Request,
+  response: Response,
+  next: NextFunction
+): void {
+  const primaryOrigin = getPrimaryHostUrl();
+  if (primaryOrigin) {
+    const origin = `${request.protocol}://${getRequestHost(request)}`;
+    if (origin !== primaryOrigin) {
```

Comment:
> I'm surprised this check involves anything more than the two hostnames

The configuration thing we have to check against is an origin (protocol + host), so either I’m doing the work here to compose the current request’s origin, or I’m doing work to extract the host from `HOST_URL` (see above, bad naming choice, the *URL* part is important because it’s not just a host). Extra stuff has to happen here one way or the other.

> And why risk bugs around trailing slashes?

What’s the risk here?

Reply:
I think as written, no trailing slash issues exist, but I was worrying about a case where someone made a change to how either `getPrimaryHostUrl` or `getRequestHost` worked such that one returned a value with a trailing slash and the other did not.

Reply:
Changed things to just deal with the host name here, and not try to also include the protocol. See https://github.com/usdigitalresponse/univaf/pull/665#discussion_r897275172